contracts/modules/v2/Assessment.sol

```diff
@@ -0,0 +1,485 @@
+
+    config.minVotingPeriodInDays = 3; // days
+    config.payoutCooldownInDays = 1; // days
+    config.stakeLockupPeriodInDays = 14; // days
+    config.silentEndingPeriodInDays = 1; // days
+  }
+
+  /* ========== VIEWS ========== */
+
+  function min(uint a, uint b) internal pure returns (uint) {
+    return a <= b ? a : b;
+  }
+
+  /// @dev Returns the vote count of an assessor.
+  ///
+  /// @param assessor  The address of the assessor.
+  function getVoteCountOfAssessor(address assessor) external override view returns (uint) {
+    return votesOf[assessor].length;
+  }
+
+  /// @dev Returns the number of assessments.
+  function getAssessmentsCount() external override view returns (uint) {
+    return assessments.length;
+  }
+
+  /// @dev Returns only the poll from the assessment struct to make only one SLOAD. Is used by
+  /// other contracts.
+  ///
+  /// @param assessmentId  The index of the assessment
+  function getPoll(uint assessmentId) external override view returns (Poll memory) {
+    return assessments[assessmentId].poll;
+  }
+
+  /// Returns all pending rewards, the withdrawable amount and the index until which they can be
+  /// withdrawn.
+  ///
+  /// @param staker  The address of the staker
+  function getRewards(address staker) external override view returns (
+    uint totalPendingAmount,
+    uint withdrawableAmount,
+    uint withdrawableUntilIndex
+  ) {
+    uint104 rewardsWithdrawableFromIndex = stakeOf[staker].rewardsWithdrawableFromIndex;
+    Vote memory vote;
+    Assessment memory assessment;
+    uint voteCount = votesOf[staker].length;
+    bool hasReachedUnwithdrawableReward = false;
+
+    for (uint i = rewardsWithdrawableFromIndex; i < voteCount; i++) {
+      vote = votesOf[staker][i];
+      assessment = assessments[vote.assessmentId];
+
+      // If hasReachedUnwithdrawableReward is true, skip and continue calculating the pending total
+      // rewards.
+      if (
+        !hasReachedUnwithdrawableReward &&
+        assessment.poll.end + config.payoutCooldownInDays * 1 days >= block.timestamp
+      ) {
+        hasReachedUnwithdrawableReward = true;
+        // Store the index of the vote until which rewards can be withdrawn.
+        withdrawableUntilIndex = i;
+        // Then, also store the pending total value that can be withdrawn until this index.
+        withdrawableAmount = totalPendingAmount;
+      }
+
+      totalPendingAmount += uint(assessment.totalReward) * uint(vote.stakedAmount) /
+        (uint(assessment.poll.accepted) + uint(assessment.poll.denied));
+    }
+
+    if (!hasReachedUnwithdrawableReward) {
+      withdrawableUntilIndex = voteCount;
+      withdrawableAmount = totalPendingAmount;
+    }
+  }
+
+  /* === MUTATIVE FUNCTIONS ==== */
+
+  /// Increases the sender's stake by the specified amount and transfers NXM to this contract
+  ///
+  /// @param amount  The amount of nxm to stake
+  function stake(uint96 amount) public whenNotPaused {
```

Comment:
check the caller is member

Reply:
This function involves an nxm transfer. You cannot hold nxm if you're not a member.